spaceship/lib/spaceship/client.rb

```diff
@@ -491,22 +576,28 @@ def send_shared_login_request(user, password)
           modified_cookie.gsub!(unescaped_important_cookie, escaped_important_cookie)
         end
 
-        # Fixes issue https://github.com/fastlane/fastlane/issues/21071
-        # On 2023-02-23, Apple added a custom implementation
-        # of hashcash to their auth flow
-        # hashcash = nil
-        hashcash = self.fetch_hashcash
-
-        response = request(:post) do |req|
-          req.url("https://idmsa.apple.com/appleauth/auth/signin")
-          req.body = data.to_json
-          req.headers['Content-Type'] = 'application/json'
-          req.headers['X-Requested-With'] = 'XMLHttpRequest'
-          req.headers['X-Apple-Widget-Key'] = self.itc_service_key
-          req.headers['Accept'] = 'application/json, text/javascript'
-          req.headers["Cookie"] = modified_cookie if modified_cookie
-          req.headers["X-Apple-HC"] = hashcash if hashcash
-        end
+        do_legacy_signin = ENV['FASTLANE_USE_LEGACY_PRE_SIRP_AUTH']
```

Comment:
A ENV to use to use existing logic _if_ it ever starts working again 🤷‍♂️ I didn't feel comfortable deleting that path